app/twitter.py

```diff
@@ -1,129 +1,205 @@
-			"text":dm.message_create['message_data']['text'],
-			"hashtags":dm.message_create['message_data']['entities']['hashtags'],
-			"symbols":dm.message_create['message_data']['entities']['symbols'],
-			"user_mentions":dm.message_create['message_data']['entities']['user_mentions'],
-			"urls":dm.message_create['message_data']['entities']['urls'],
-			})
-	return dm_list
 
+def create_welcome_message(name: str,
+                           msg_txt: str,
+                           quick_replies: List[Dict] = None):  # Use quick_reply_option
+    manual_twitter_auth = manual_twitter_api()
+    url = "https://api.twitter.com/1.1/direct_messages/welcome_messages/new.json"
+    if quick_replies:
+        payload = json.dumps({
+            "welcome_message": {
+                "name": name,
+                "message_data": {
+                    "text": msg_txt,
+                    "quick_reply": {
+                        "type": "options",
+                                "options": quick_replies
+                    }
+                }
+            }
+        })
+    else:
+        payload = json.dumps({
+            "welcome_message": {
+                "name": name,
+                "message_data": {
+                    "text": msg_txt
+                }
+            }
+        })
+    headers = {
+        'Content-Type': 'application/json'
+    }
+
+    response = manual_twitter_auth.post(url, headers=headers, data=payload)
+
+    welcome_response = json.loads(response.text)
+    print(welcome_response)
+
+
+def send_dm(user_id: str, dm_body: str) -> str:
+    """ Function to send dm to a given user_id """
+    api = create_api()
+    dm = api.send_direct_message(user_id, dm_body)
+    return dm
+
+
+def process_dms(user_id: str, tweet_id: str, convo_tree_txt: str) -> Dict:
+    """Function to get response DMs sent from button in tweet"""
+    api = create_api()
+    dms = api.list_direct_messages()
+    dm_list = []
+    for dm in dms:
+        if dm.message_create['sender_id'] == user_id:
+            # Get response preform if
+            if dm.message_create['message_data']['quick_reply_response']['metadata'] == 'confirm_yes':
+                # Make the form link using the user_id
+                form_link = f'https://a.humanrightsfirst.dev/edit/{tweet_id}'
+                response_txt = convo_tree_txt + '\n' + form_link
+                # Send form link to user
+                api.send_direct_message(user_id, response_txt)
+            else:
+                # Send final response to user
+                api.send_direct_message(user_id, convo_tree_txt)
+            return {
+                "reachout_template": dm.initiated_via['welcome_message_id'],
+                "tweeter_id": dm.message_create['sender_id'],
+                "dm_text": dm.message_create['message_data']['text'],
+                "quick_reply_response": dm.message_create['message_data']['quick_reply_response']['metadata']
+            }
```

Comment:
Proper use of UPER! This is great code. I instantly understand what's going on.

Glad we're finally getting this form implemented.